Content.Server/Botany/Systems/PlantAnalyzerSystem.cs

```diff
@@ -0,0 +1,261 @@
+
+        args.Handled = true;
+        _cancellationTokenSources.Remove(component);
+    }
+
+    private void OpenUserInterface(EntityUid user, EntityUid analyzer)
+    {
+        if (!TryComp<ActorComponent>(user, out var actor) || !_uiSystem.TryGetUi(analyzer, PlantAnalyzerUiKey.Key, out var ui))
+            return;
+
+        _uiSystem.OpenUi(ui, actor.PlayerSession);
+    }
+
+    public void UpdateScannedUser(EntityUid uid, EntityUid user, EntityUid target, Entity<PlantAnalyzerComponent> component)
+    {
+        if (!_uiSystem.TryGetUi(uid, PlantAnalyzerUiKey.Key, out var ui))
+            return;
+
+        TryComp<PlantHolderComponent>(target, out var plantcomp);
+        TryComp<SeedComponent>(target, out var seedcomponent);
+
+        var state = default(PlantAnalyzerScannedSeedPlantInformation);
+        var seedData = default(SeedData); // Data of a unique Seed.
+        var seedProtoId = default(SeedPrototype); // Data of base seed prototype.
+
+        if (seedcomponent != null)
+        {
+            if (seedcomponent.Seed != null) // If unique seed.
+            {
+                seedData = seedcomponent.Seed;
+                state = ObtainingGeneDataSeed(seedData, target, false, component.Comp.AdvancedScan);
+            }
+            else if (seedcomponent.SeedId != null && _prototypeManager.TryIndex(seedcomponent.SeedId, out SeedPrototype? protoSeed)) // Get the seed protoype.
+            {
+                seedProtoId = protoSeed;
+                state = ObtainingGeneDataSeedProt(protoSeed, target, component.Comp.AdvancedScan);
+            }
+        }
+        else if (plantcomp != null) // If we poke the plantholder, it checks the plantholder seed.
+        {
+            seedData = plantcomp.Seed;
+            if (seedData != null)
+            {
+                state = ObtainingGeneDataSeed(seedData, target, true, component.Comp.AdvancedScan);  // SeedData is a unique seed in a tray.
+            }
+        }
+
+        if (state == null)
+            return;
+
+        _uiSystem.SendUiMessage(ui, state);
+    }
+
+    /// <summary>
+    ///     Analysis of seed from prototype.
+    /// </summary>
+    public PlantAnalyzerScannedSeedPlantInformation ObtainingGeneDataSeedProt(SeedPrototype comp, EntityUid target, Boolean scanMode)
+    {
+        var seedName = "\n  " + Loc.GetString("plant-analyzer-window-label-trademark") +
+            "\n    " + Loc.GetString(comp.DisplayName);
+
+        var seedChem = "\n   ";
+        var plantHarvestType = "";
+        var exudeGases = "";
+
+        if (comp.HarvestRepeat == HarvestType.Repeat) plantHarvestType = Loc.GetString("plant-analyzer-harvest-repeat");
+        if (comp.HarvestRepeat == HarvestType.NoRepeat) plantHarvestType = Loc.GetString("plant-analyzer-harvest-ephemeral");
+        if (comp.HarvestRepeat == HarvestType.SelfHarvest) plantHarvestType = Loc.GetString("plant-analyzer-harvest-autoharvest");
```

Comment:
Locstrings should be resolved on the client instead of sending already localized strings over the network.